### PR TITLE
Partitioned file naming

### PR DESCRIPTION
**Pull request by lverdoes**
This PR introduces the option to name the output files of a partition statement. The flag `FILEFORMAT` is added to the `COPY` statement and accepts a format string that forms the basis of the final output file.

The format string accepts the option to insert an index with: `{i}`, or a UUID: `{uuid}`.
For example:
`COPY 'table' TO 'dir' (PARTITION_BY (column0), OVERWRITE_OR_IGNORE, FILEFORMAT "data_{uuid}");`
will create one or more files in 'dir' with unique names and prefixed with 'data_'

This behaviour allows for an append workflow as you can keep adding files to an existing directory. The ALLOW_OVERWRITE flag is replaced with OVERWRITE_OR_IGNORE

If `FILEFORMAT` is not specified - or left empty - the output name defaults to 'data_' + an index.

@samansmink

### CODE REVIEW

**Review by samansmink — COMMENTED** [review]
thanks for the pr @lverdoes looks good! i added a few comments

File: src/planner/binder/statement/bind_copy.cpp [hunk 1]
```diff
@@ -92,11 +94,17 @@ BoundStatement Binder::BindCopyTo(CopyStatement &stmt) {
...
 			continue;
 		}
-		if (loption == "allow_overwrite") {
-			allow_overwrite =
+		if (loption == "overwrite_or_ignore") {
+			overwrite_or_ignore =
 			    option.second.empty() || option.second[0].CastAs(context, LogicalType::BOOLEAN).GetValue<bool>();
 			continue;
 		}
+		if (loption == "filenamepattern") {
```
Comment (samansmink) [inline src/planner/binder/statement/bind_copy.cpp:102]: can we call it `filename_pattern`?

**Review by samansmink — COMMENTED** [review]
final set of nitpicks! the rest looks good :)

**Comment by Mytherin** [comment]
@samansmink is this ready to be merged? The regression tests are failing, but I suppose that might be because the new flag isn’t supported in the current master?

**Comment by samansmink** [comment]
@Mytherin yea thats correct, this is good to go!

**Comment by djouallah** [comment]
ok, it seems this behavior was changed recently, it seems OVERWRITE_OR_IGNORE 1 will delete existing files regardless if the name match or not ? i used this pattern to have an append workflow

**Pull request merged**